Upstream-Status: Inappropriate

diff --git a/board/freescale/imx8mq_evk/imx8mq_evk.c b/board/freescale/imx8mq_evk/imx8mq_evk.c
index 607170e..d4c3262 100644
--- a/board/freescale/imx8mq_evk/imx8mq_evk.c
+++ b/board/freescale/imx8mq_evk/imx8mq_evk.c
@@ -33,7 +33,7 @@
 
 DECLARE_GLOBAL_DATA_PTR;
 
-#define UART_PAD_CTRL	(PAD_CTL_DSE6 | PAD_CTL_FSEL1)
+#define UART_PAD_CTRL	(PAD_CTL_DSE6 | PAD_CTL_FSEL1 | PAD_CTL_PUE)
 
 #define WDOG_PAD_CTRL	(PAD_CTL_DSE6 | PAD_CTL_HYS | PAD_CTL_PUE)
 
@@ -300,6 +300,9 @@ int board_late_init(void)
 	board_late_mmc_env_init();
 #endif
 
+	puts("DeviceName is ECU150A1.\n");
+	env_set("devicename", "ecu150a1");
+	env_set("bootargs_ram_capacity", "cma=384M");
 	return 0;
 }
 
